frontend/src/components/listeners/ExtractorStatus.tsx

```diff
@@ -69,10 +72,27 @@ function Row (props: { summary: any; updates: any; }) {
 	);
 }
 
-export default function ExtractorStatus(props: { job_id: any; summary: any; updates: any; }) {
+export default function ExtractorStatus(props: { job_id: any; }) {
 	const job_id = props.job_id;
-	const summary = props.summary;
-	const updates = props.updates;
+
+	const dispatch = useDispatch();
+
+	const jobSummaryFetch = (job_id: string | undefined) => dispatch(fetchJobSummary(job_id));
+	const jobUpdatesFetch = (job_id: string | undefined) => dispatch(fetchJobUpdates(job_id));
+
+	const summary = useSelector((state: RootState) => state.listener.currJobSummary);
+	const updates = useSelector((state: RootState) => state.listener.currJobUpdates);
+
+	useEffect(() => {
+		const interval = setInterval(() => {
+			if (job_id.length > 0) {
+				jobSummaryFetch(job_id);
+				jobUpdatesFetch(job_id);
+			}
+		}, 2000);
+
+		return () => clearInterval(interval);
+	}, [job_id]);
```

Comment:
Ah i see. You need to move the timer logic into the status component instead of the parent component. Neat!